File: 2020/day_1/problem_one.py
def main():
    numbers = number_iter("input.txt")
    answer = solve_problem(numbers)
    print(answer)

def number_iter(filename):
    # Returns an iterator of ints in a file containing new line separated ints
    with open(filename) as file:
        lines = file.readlines()
    return map(lambda x: int(x), lines)

def solve_problem(numbers, total=2020):
    # Iterates through numbers, attempts to find a match that adds to sum, and then returns the multiple of those numbers
    # Using a dictionary method because that seemed neat
    seen = dict()
    for num in numbers:
        seen[num] = 0
        if (total - num) in seen:
            # Found a match
            return (total - num) * num
    # Didn't find anything
    return -1

if __name__ == "__main__":
    main()

File: 2018/day_1/src/bin/problem_one.rs
use std::fs;
use std::io::{self, BufRead};

fn main() {
    let file = fs::File::open("input.txt").expect("Invalid Filename");
    let lines = io::BufReader::new(file)
        .lines();
    
    let result: i64 = lines
        .map(|x| x.unwrap().parse::<i64>().unwrap())
        .sum();
    
    println!("{}", result);
}
